src/runtime/runtime_wasm_js.go

```diff
@@ -2,26 +2,15 @@
 
 package runtime
 
-import "unsafe"
-
 type timeUnit float64 // time in milliseconds, just like Date.now() in JavaScript
 
 // wasmNested is used to detect scheduler nesting (WASM calls into JS calls back into WASM).
 // When this happens, we need to use a reduced version of the scheduler.
+//
+// TODO: this variable can probably be removed once //go:wasmexport is the only
```

Comment:
Should the other mechanism for calling an export in js use the same semantics?

Reply:
I find it difficult to say. `//export` currently doesn't have clearly documented semantics around the scheduler.
The only thing I was able to find was this comment in https://github.com/tinygo-org/tinygo/pull/2011:

> If WASM calls into JS which calls back into WASM, it is possible for the scheduler to nest.
> The event from the callback must be handled immediately, so the task cannot simply be deferred to the outer scheduler.
> This creates a minimal scheduler loop which is used to handle such nesting.

However, all tests still pass with this change. So it's not documented, and there are no tests. I don't even know what the correct behavior is supposed to be exactly. So, uh, :woman_shrugging:
My suggestion would be to leave this alone, and if it causes any issues we can fix them when they happen.